lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -496,11 +504,11 @@ def extend(self, other):
         if len(self):
             new_kymos = set([track._kymo._id for track in other])
             assert (
-                len(new_kymos) == 1 and self[0]._kymo._id == list(new_kymos)[0]
+                len(new_kymos) == 1 and self._kymo._id == list(new_kymos)[0]
             ), "All tracks must have the same source kymograph."
             new_channels = set([track._channel for track in other])
             assert (
-                len(new_channels) == 1 and self[0]._channel == list(new_channels)[0]
+                len(new_channels) == 1 and self._channel == list(new_channels)[0]
             ), "All tracks must be from the same color channel."
```

Comment:
Would it make sense to have one central function like `check_kymo_compatiblity()`, which can be called from here and the `KymoTrackGroup.__init__()`? This would deduplicate the code to check if kymos are compatible.